### PR TITLE
Release `v6.5.0`

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: pallet/staking/src/migration.rs [hunk 1]
```diff
@@ -125,10 +125,10 @@ pub mod v1 {
 
 			// Check that everything decoded fine.
 			<ExposureCache1<T>>::iter_keys().for_each(|k| {
-				assert!(<ExposureCache0<T>>::try_get(k).is_ok(), "Can not decode V1 `Exposure`.");
+				assert!(<ExposureCache1<T>>::try_get(k).is_ok(), "Can not decode V1 `Exposure`.");
 			});
 			<ExposureCache2<T>>::iter_keys().for_each(|k| {
-				assert!(<ExposureCache0<T>>::try_get(k).is_ok(), "Can not decode V1 `Exposure`.");
+				assert!(<ExposureCache2<T>>::try_get(k).is_ok(), "Can not decode V1 `Exposure`.");
```
Comment (boundless-forest) [inline pallet/staking/src/migration.rs:131]: Can not decode V2 `Exposure`

Comment (AurevoirXavier) [reply pallet/staking/src/migration.rs:131]: `v0 -> v1`

**Review by boundless-forest — DISMISSED** [review]

**Comment by AurevoirXavier** [comment]
## Darwinia try-runtime
```rs
[2023-11-17T09:00:50Z INFO  remote-ext] since no at is provided, setting it to latest finalized head, 0xaa1ae84d8f6c604f17e92d70c9ee9e77cdd6008b5b9a1c3df050842dad609794
[2023-11-17T09:00:50Z INFO  remote-ext] since no prefix is filtered, the data for all pallets will be downloaded
[2023-11-17T09:00:50Z INFO  remote-ext] scraping key-pairs from remote at block height 0xaa1ae84d8f6c604f17e92d70c9ee9e77cdd6008b5b9a1c3df050842dad609794
✅ Found 24575 keys (1.84s)
[00:00:23] ✅ Downloaded key values 1,050.6608/s [========================================================================================================================================================================================] 24575/24575 (0s)
✅ Inserted keys into DB (0.09s)
[2023-11-17T09:01:15Z INFO  remote-ext] adding data for hashed prefix: , took 25.33s
[2023-11-17T09:01:15Z INFO  remote-ext] adding data for hashed key: 3a636f6465
[2023-11-17T09:01:16Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8
[2023-11-17T09:01:16Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac
[2023-11-17T09:01:16Z INFO  remote-ext] 👩‍👦 no child roots found to scrape
[2023-11-17T09:01:16Z INFO  remote-ext] initialized state externalities with storage root 0x0a2f48f0c2502c843d92bdfb7a7d6ee4ae0c92a3051584ca7790b0fe8017a494 and state_version V0
[2023-11-17T09:01:16Z INFO  try-runtime::cli] Original runtime [Name: RuntimeString::Owned("Darwinia2")] [Version: 6404] [Code hash: 0x5fd2...f577]
[2023-11-17T09:01:16Z INFO  try-runtime::cli] New runtime      [Name: RuntimeString::Owned("Darwinia2")] [Version: 6500] [Code hash: 0x1edb...6532]
[2023-11-17T09:01:16Z INFO  try-runtime::cli] 🚀 Speed up your workflow by using snapshots instead of live state. See `try-runtime create-snapshot --help`.
[2023-11-17T09:01:16Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 Running TryRuntime_on_runtime_upgrade with checks: All
[2023-11-17T09:01:17Z INFO  darwinia_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:17Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:17Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 TryRuntime_on_runtime_upgrade succeeded! Running it again without checks for weight measurements.
[2023-11-17T09:01:17Z INFO  darwinia_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:17Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:17Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 Running TryRuntime_on_runtime_upgrade again to check idempotency: All
[2023-11-17T09:01:17Z INFO  darwinia_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:17Z WARN  darwinia_staking::migration::v1] [pallet::staking] skipping v0 to v1 migration: executed on wrong storage version. Expected version 0, found StorageVersion(1)
[2023-11-17T09:01:17Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:17Z INFO  try-runtime::cli] PoV size (zstd-compressed compact proof): 13.6 KB. For parachains, it's your responsibility to verify that a PoV of this size fits within any relaychain constraints.
[2023-11-17T09:01:17Z INFO  try-runtime::cli] Consumed ref_time: 0.005225s (1.04% of max 0.5s)
[2023-11-17T09:01:17Z INFO  try-runtime::cli] ✅ No weight safety issues detected. Please note this does not guarantee a successful runtime upgrade. Always test your runtime upgrade with recent state, and ensure that the weight usage of your migrations will not drastically differ between testing and actual on-chain execution.
```

**Comment by AurevoirXavier** [comment]
## Crab try-runtime
```rs
[2023-11-17T09:00:32Z INFO  remote-ext] since no at is provided, setting it to latest finalized head, 0x9b04a0c6cda403e2ea7f23f7ef32b22c09eedd887ad9c055822e8f44887754bb
[2023-11-17T09:00:32Z INFO  remote-ext] since no prefix is filtered, the data for all pallets will be downloaded
[2023-11-17T09:00:32Z INFO  remote-ext] scraping key-pairs from remote at block height 0x9b04a0c6cda403e2ea7f23f7ef32b22c09eedd887ad9c055822e8f44887754bb
✅ Found 260438 keys (18.56s)
[00:00:34] ✅ Downloaded key values 7,637.3979/s [======================================================================================================================================================================================] 260438/260438 (0s)
✅ Inserted keys into DB (0.53s)
[2023-11-17T09:01:26Z INFO  remote-ext] adding data for hashed prefix: , took 53.37s
[2023-11-17T09:01:26Z INFO  remote-ext] adding data for hashed key: 3a636f6465
[2023-11-17T09:01:26Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8
[2023-11-17T09:01:26Z INFO  remote-ext] adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef702a5c1b19ab7a04f536c519aca4983ac
[2023-11-17T09:01:26Z INFO  remote-ext] 👩‍👦 no child roots found to scrape
[2023-11-17T09:01:26Z INFO  remote-ext] initialized state externalities with storage root 0x2854f42126cf21629c5a12b315b72c6070dcc1974727fa0245359b3222e23c1b and state_version V0
[2023-11-17T09:01:27Z INFO  try-runtime::cli] Original runtime [Name: RuntimeString::Owned("Crab2")] [Version: 6404] [Code hash: 0x0fea...cdeb]
[2023-11-17T09:01:27Z INFO  try-runtime::cli] New runtime      [Name: RuntimeString::Owned("Crab2")] [Version: 6500] [Code hash: 0x0cd9...4649]
[2023-11-17T09:01:27Z INFO  try-runtime::cli] 🚀 Speed up your workflow by using snapshots instead of live state. See `try-runtime create-snapshot --help`.
[2023-11-17T09:01:27Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 Running TryRuntime_on_runtime_upgrade with checks: All
[2023-11-17T09:01:27Z INFO  crab_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:27Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:27Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 TryRuntime_on_runtime_upgrade succeeded! Running it again without checks for weight measurements.
[2023-11-17T09:01:27Z INFO  crab_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:27Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:27Z INFO  try_runtime_core::commands::on_runtime_upgrade] 🔬 Running TryRuntime_on_runtime_upgrade again to check idempotency: All
[2023-11-17T09:01:27Z INFO  crab_runtime] try-runtime::on_runtime_upgrade
[2023-11-17T09:01:27Z WARN  darwinia_staking::migration::v1] [pallet::staking] skipping v0 to v1 migration: executed on wrong storage version. Expected version 0, found StorageVersion(1)
[2023-11-17T09:01:27Z INFO  runtime::frame-support] ⚠️ Ethereum declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `NoStorageVersionSet`
[2023-11-17T09:01:27Z INFO  try-runtime::cli] PoV size (zstd-compressed compact proof): 8.8 KB. For parachains, it's your responsibility to verify that a PoV of this size fits within any relaychain constraints.
[2023-11-17T09:01:27Z INFO  try-runtime::cli] Consumed ref_time: 0.0041s (0.82% of max 0.5s)
[2023-11-17T09:01:27Z INFO  try-runtime::cli] ✅ No weight safety issues detected. Please note this does not guarantee a successful runtime upgrade. Always test your runtime upgrade with recent state, and ensure that the weight usage of your migrations will not drastically differ between testing and actual on-chain execution.
```

**Pull request merged**